File: distros/A/Affix/eg/Cookbook/array_reverse.c
int array_reverse(int a[], int len) {
    int tmp, i;
    int ret = 0;
    for (i = 0; i < len / 2; i++) {
        ret += a[i];
        tmp = a[i];
        a[i] = a[len - i - 1];
        a[len - i - 1] = tmp;
    }
    return ret;
}

int array_reverse10(int a[10]) {
    return array_reverse(a, 10);
}

File: distros/A/Affix/eg/Cookbook/array_reverse.pl
use Affix;
use strict;
$|++;
affix './array_reverse.so', array_reverse => [ ArrayRef [Int], Int ], Int;
affix './array_reverse.so', array_reverse10 => [ ArrayRef [ Int, 10 ] ], Int;
#
my @a = ( 1 .. 10 );
array_reverse10( \@a );
print "$_ " for @a;
print "\n";
#
@a = ( 1 .. 20 );
array_reverse( \@a, 20 );
print "$_ " for @a;
print "\n";
#
my $a = [ 1 .. 20 ];
array_reverse( $a, 20 );
print "$_ " for @$a;
print "\n";
